### ISSUE TITLE
Audio padding in collation produces inconsistent pad length

### ISSUE THREAD

**Issue by janvainer** [lhotse-speech/lhotse#218]
The following test fails:

```
def test_collate_audio_padding():
    cuts = CutSet.from_json('test/fixtures/ljspeech/cuts.json')
    assert len(set(cut.num_samples for cut in cuts)) > 1
    correct_num_samples = max(cut.num_samples for cut in cuts)
    cuts = cuts.pad()
    assert next(iter(cuts)).num_samples == correct_num_samples
```

**Comment by pzelasko** [comment]
I looked into this test. It might be counter-intuitive at first, but the behavior is correct (i.e. as I intended it - I'm open to discussing and revising it though).

Basically, there are three ways to represent time:
- duration (float, seconds)
- samples (int, count)
- frames (int, count)

None of those is universal and they also cannot be 100% consistent by construction (when you convert samples -> frames you will have some discontinuities at the signal's end, e.g. zero pad with samples, etc.; if you compute all operations on float seconds, you will run in-between frames/samples + precision issues, etc.).

So the way I solve that in padding is: if there are features attached to  cut, it means you're interested in frames, and they take precedence, so everything is padded so that num_samples is an integer multiplier of num_frames (if you load audio, we'll pad it with zeros at the end). If there are not features, it means you're interested mostly in samples, and we'll pad to num_samples directly; if none of those are available then I think it's actually a usage error but we'll use duration then.
